dflib/src/main/java/org/dflib/Exp.java

```diff
@@ -88,23 +89,71 @@ public interface Exp<T> {
     }
 
     /**
-     * Returns a DateExp whose "eval" returns a Series with the value argument at each position, and "reduce"
+     * Returns a {@code NumExp<Integer>} whose "eval" returns a Series with the value argument at each position, and "reduce"
+     * returns the value itself.
+     */
```

Comment:
I suggest instead of `XyzScalarExp` to return `NumExp<Xyz>` . For `BooleanScalarExp`, return `Condition`. I.e. hide the implementation and stick to the common interface in the method signature.